apis/concierge/authentication/v1alpha1/types_tls.go.tmpl

```diff
@@ -3,9 +3,24 @@
 
 package v1alpha1
 
+// CABundleSource provides a source for CA bundle used for client-side TLS verification.
+type CABundleSource struct {
+	// Whether the CA bundle is being sourced from a kubernetes secret or a configmap.
+	// +kubebuilder:validation:Enum=Secret;ConfigMap
+	Kind string `json:"kind,omitempty"`
+	// Name of the secret or configmap from which to read the CA bundle.
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
+	// Key within the secret or configmap from which to read the CA bundle.
+	// +kubebuilder:validation:MinLength=1
+	Key string `json:"key,omitempty"`
```

Comment:
Do we want a default on this one? Like maybe `ca.crt`. Or would we rather force the user to always specify this one?

Reply:
It would be better to have the user always specify this. Only cert-manager uses this key in its secret, but only when it decides to include the CA certificate in the certificate it reconciles.